### ISSUE TITLE
Can't compile because of `packed_simd` version 0.3.3

### ISSUE THREAD

**Issue by pengye91** [ballista-compute/ballista#469]
Hi guys, the dependency of `packed_simd v0.3.3` seemed broken on crates.io[0]

When I follow the user guide, it can't compile because of `packed_simd v0.3.3`: 

```
error[E0432]: unresolved import `crate::arch::x86_64::_mm_shuffle_pi8`
   --> /Users/yxie/.cargo/registry/src/github.com-1ecc6299db9ec823/packed_simd-0.3.3/src/codegen/shuffle1_dyn.rs:40:29
    |
40  |                         use crate::arch::x86_64::_mm_shuffle_pi8;
    |                             ^^^^^^^^^^^^^^^^^^^^^---------------
    |                             |                    |
    |                             |                    help: a similar name exists in the module: `_mm_shuffle_epi8`
    |                             no `_mm_shuffle_pi8` in `arch::x86_64`
```
...
```
error: aborting due to 7 previous errors

For more information about this error, try `rustc --explain E0432`.
error: could not compile `packed_simd`
```

And it seems that the original `packed_simd` package will no longer be updated, they choose to use another package name `packed_simd_2`[1] instead.

So do we have plans to update this dependency? 

[0] https://github.com/rust-lang/packed_simd#the-cratesio-version-can-no-longer-be-updated
[1] https://crates.io/crates/packed_simd_2

**Comment by andygrove** [comment]
I don't see this error when I compile and I am not sure why, but others have reported this problem as well. Are you specifically enabling the simd feature when you build Ballista?

**Comment by joshuarobinson** [comment]
I am getting this same error and I followed the instructions here: https://ballistacompute.org/docs/execute-query-rust.html

**Comment by djKooks** [comment]
Seems it is issue of `packed_sims` version which has been used in previous `arrow` package.
Current snapshot version seems working(I've imported from main branch source):
```
[dependencies]
ballista = { path = "/Users/me/workspace/ballista/rust/ballista" }
```

```
$ cargo build
...
cargo build
   Compiling tokio v1.2.0
   Compiling tokio-util v0.6.3
   Compiling tokio-stream v0.1.3
   Compiling datafusion v4.0.0-SNAPSHOT (https://github.com/apache/arrow?rev=b18c567#b18c5677)
   Compiling tower v0.4.4
   Compiling h2 v0.3.0
   Compiling hyper v0.14.4
   Compiling tonic v0.4.0
   Compiling arrow-flight v4.0.0-SNAPSHOT (https://github.com/apache/arrow?rev=b18c567#b18c5677)
   Compiling ballista v0.4.0-SNAPSHOT (/Users/me/workspace/ballista/rust/ballista)
...
```

**Comment by andygrove** [comment]
The best bet would be to start using version 0.4.0-SNAPSHOT (latest in main branch). I don't think it is worth spending time to fix 0.3.x right now.

**Comment by pengye91** [comment]
Yep, it can compile if using `0.4.0-SNAPSHOT`.

I think we should update the https://ballistacompute.org/docs/execute-query-rust.html page which may confuse newcomers.

**Comment by andygrove** [comment]
This was an issue with 0.3.0 that is now resolved in 0.4.0-SNAPSHOT and there will be an alpha release today, so closing this.
